src/frontend/src/handler/create_table.rs

```diff
@@ -1985,16 +2025,41 @@ mod tests {
                     ),
                     is_hidden: false,
                 },
+                ColumnCatalog {
+                    column_desc: ColumnDesc::from_field_with_column_id(
+                        &Field::with_name(
+                            StructType::new([("f1", DataType::Int32)]).into(),
+                            "nested",
+                        ),
```

Comment:
Yet another example of why `Field::sub_fields` (and `ColumnDesc::field_descs`) shall be [consolidated](https://github.com/risingwavelabs/risingwave/compare/main...refactor-common-fields-and-struct-WIP-r1) into `StructType`. Currently `Field::with_name` on `struct` produces incomplete results - and there is a special `Field::with_struct`.

No need to fix the usage in this PR as the incomplete results are NEVER used except `describe` #17128